core/src/main/java/org/apache/brooklyn/core/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -800,7 +809,11 @@ private boolean attemptType(String key, CatalogItemType candidateCiType) {
                     // (when we're not given a key, the previous block should apply)
                     errors.add(e);
                 } else {
-                    // all other cases, the error is probably due to us not getting the type right, ignore it
+                    // all other cases, the error is probably due to us not getting the type right, probably ignore it
+                    // but cache it if we've checked entity, we'll use that as fallback errors
```

Comment:
we're parsing CAMP, then CAMP refers to something in catalog.  the code before this change would use camp-specific logic to handle the catalog entry making use of the parent entity's class loading context if needed.  but that breaks if it refers to a non-camp non-java catalog item (eg tosca).  so this now also looks it up using the catalog mechanism (although in that case it does not have any default class loading context.)

the CLC should only matter for java types.  it's working now, but would like to clean this whole area up massively!